### PR TITLE
fix: trim new line for azure storage account name

### LINKED ISSUE #63326: AzureFile Mount fails - putting new line between storage account and domain
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
 /kind bug
> /kind feature

**What happened**:
When trying to mount an Azure File mount it fails, it looks as though its putting a new line between the name of the Azure Storage Account and the Azure storage domain. There are no new line characters in the secret that holds the account name.

The following is the secret:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: asa-otm
  labels:
    app: my-app
    role: secret
    env: dev
type: Opaque
data:
  azurestorageaccountname: c2FhdXNucHJkZGV2b3RtCg==
  azurestorageaccountkey: PASSWORD  
```

The following is the declaration within the `container` section of the Kubernetes yaml:
```yaml
        volumeMounts:
        - name: asa-otm
          mountPath: /opt/my-app/otm
      volumes:
      - name: asa-otm
        azureFile:
          secretName: asa-otm
          shareName: otm
          readOnly: false
```

The following is the error that's shown in `kubectl describe po/pod_name:
```
Warning  FailedMount            11s                kubelet, k8s-agentpool0-18189144-2  MountVolume.SetUp failed for volume "asa-otm" : mount failed: exit status 1
Mounting command: mount
Mounting arguments: -t cifs -o username=saausnprddevotm
,password=PASSWORD
,file_mode=0755,dir_mode=0755,vers=3.0 //saausnprddevotm
.file.core.windows.net/otm /var/lib/kubelet/pods/3ae17688-4cdd-11e8-9240-000d3ae0c5ae/volumes/kubernetes.io~azure-file/asa-otm
Output: mount error: could not resolve address for saausnprddevotm
.file.core.windows.net: Unknown error
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): `v1.9.4`
- Cloud provider or hardware configuration: azure
- OS (e.g. from /etc/os-release): Ubuntu  16.04.3 LTS
- Kernel (e.g. `uname -a`): 4.13.0-1011-azure
- Install tools: [acs-engine](https://github.com/Azure/acs-engine)
- Others: [acs-engine #2818](https://github.com/Azure/acs-engine/issues/2818)

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
trim new line for azure storage account name, new line is not allowed in azure storage account name, and sometimes user may input new line by accident:
```
echo blabla | base64
```

**Which issue(s) this PR fixes**:

Fixes #63326

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
none
```

/kind bug
/assign @feiskyer 
/priority important-soon
/sig cloud-provider
/area provider/azure

cc @adelwin

### CODE REVIEW

File: pkg/volume/azure_file/azure_util.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ func (s *azureSvc) GetAzureCredentials(host volume.VolumeHost, nameSpace, secret
 	if accountName == "" || accountKey == "" {
 		return "", "", fmt.Errorf("Invalid %v/%v, couldn't extract azurestorageaccountname or azurestorageaccountkey", nameSpace, secretName)
 	}
+	accountName = strings.TrimSuffix(accountName, "\n")
```
Comment (feiskyer) [inline pkg/volume/azure_file/azure_util.go]: Use [strings.TrimSpace](https://golang.org/pkg/strings/#TrimSpace) instead

Comment (andyzhangx) [reply pkg/volume/azure_file/azure_util.go]: fixed, PTAL, thanks.

**Review by feiskyer — APPROVED** [review]
/lgtm
/approve

**Pull request merged**